addons/hack/fakegen.go

```diff
@@ -0,0 +1,12 @@
+package hack
+
+import (
+	crtclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+//go:generate counterfeiter -o ../pkg/fakeclusterclient/crtclusterclient.go --fake-name CRTClusterClient . CrtClient
```

Comment:
Also wouldn't it be better for a ```fakes``` directory be created in order to group of all future addons fake files (instead of current ```fakeclusterclient```) and better code readability:  e.g.```../pkg/fakes/crtclusterclient.go```?

Reply:
I think we can restructure if we have more fakes in the future